lib/ecto/adapters/sql.ex

```diff
@@ -983,6 +995,12 @@ defmodule Ecto.Adapters.SQL do
       ?\s,
       inspect(params, charlists: false)
     ]
+    if stacktrace_match_path = Application.get_env(:ecto_sql, :log_stacktrace_match_path) do
```

Comment:
This should be stored in the `opts`, as opposed to application env. This is a better practice for configurability.

Reply:
Also, I'm wondering if there's a better approach instead of matching the first result into selected path. Is there any other option you've considered?

Reply:
@v0idpwn
Thanks for review.
 > This should be stored in the opts, as opposed to application env. This is a better practice for configurability.

I changed it to `log_stacktrace_match_path = Keyword.get(opts, :log_stacktrace_match_path) || Keyword.get(opts, :otp_app) |> to_string()` and this logging to default output. Is it bad to have the default output?

> Also, I'm wondering if there's a better approach instead of matching the first result into selected path. Is there any other option you've considered?

Because I referred to [the Rails implementation](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/log_subscriber.rb#L139), and it's often useful.
Or output with all match stack trace? But I think that will became noisy log.